### PR TITLE
OGW: compute hmax from elevations

### PR DESCRIPTION
**Pull request by jiahe23**
We've decided to split #1569 into two separate task: 1) to compute hmax, 2) to compute T tensor.

This PR adds the computation for hmax (used in orographic gravity wave) from elevation (right now it is read from gfdl restart file).

Please be noted that this PR only won't enable the feature of directly computing OGW related topo info from elevation, but I included the code structure changes and will test it once the second PR (T tensor calculation) is ready. I have left a note in the ogw parameterization setup as well.

The hmax calculation has been tested thoroughly.

### CODE REVIEW

**Comment by jiahe23** [comment]
hr_hmax_4min_100e3
Comparison of hmax from the gfdl restart (upper) and our calculation from a 4min resolution elevation (bottom).

**Comment by jiahe23** [comment]
coarse_hmax_100e3
Comparison of hmax from the gfdl restart (upper) and our calculation from the elevation used in current Earth wrap setup(bottom).

File: regression_tests/mse_tables.jl
Comment (szy21) [inline regression_tests/mse_tables.jl:1]: Why do we need to change this file?

File: src/parameterizations/gravity_wave_drag/orographic_gravity_wave_helper.jl [hunk 1]
```diff
@@ -1,5 +1,57 @@
...
+    ilon_range =
+        min.(
+            Int.(round.(scale ./ (deg2rad(dlon) .* cosd.(lat)))),
+            Int(round(length(lon) / 16)),
+        )
+
+    hmax = zeros(size(elev))
+    for (i, ilon) in enumerate(lon)
+        if i % 10 == 1
+            println("i = $i, and ilon = $ilon")
```
Comment (szy21) [inline src/parameterizations/gravity_wave_drag/orographic_gravity_wave_helper.jl:36]: Is this print intentional?

File: src/utils/cli_options.jl [hunk 2]
```diff
@@ -332,9 +332,8 @@ function argparse_settings()
         arg_type = Bool
         default = false
         "--orographic_gravity_wave"
-        help = "Apply parameterization for orographic drag on horizontal mean flow"
-        arg_type = Bool
-        default = false
+        help = "Orographic drag on horizontal mean flow [`nothing` (default), `gfdl_restart`, `elevation`]"
```
Comment (szy21) [inline src/utils/cli_options.jl:335]: `raw_topo` may be a better name but up to you:)

**Review by szy21 — APPROVED** [review]
Thanks for wrapping this up!

File: src/parameterizations/gravity_wave_drag/orographic_gravity_wave_helper.jl (hunk 1 above)
Comment (jiahe23) [reply src/parameterizations/gravity_wave_drag/orographic_gravity_wave_helper.jl:36]: oops.. for debugging purposes only. good catch!

File: regression_tests/mse_tables.jl
Comment (jiahe23) [reply regression_tests/mse_tables.jl:1]: A buildkite failed due to mse... but I don't know why when I copy it doesn't change anything... Let me double check

**Pull request merged**